assets/javascripts/test_result.js

```diff
@@ -578,14 +578,96 @@ function setupResult(jobid, state, result, status_url) {
   setInfoPanelClassName(state, result);
 }
 
-function loadEmbeddedLogFiles() {
+function delay(callback, ms) {
+  var timer = 0;
+  return function () {
+    const input = this;
+    const args = arguments;
+    clearTimeout(timer);
+    timer = setTimeout(function () {
+      callback.apply(input, args);
+    }, ms || 0);
+  };
+}
+
+function filterLogLines(input) {
+  if (input === undefined) {
+    return;
+  }
+  const string = input.value;
+  let regex = undefined;
+  const match = string.match(/^\/(.*)\/([i]*)$/);
+  if (match) {
+    regex = new RegExp(match[1], match[2]);
+  }
+  displaySearchInfo('Searching...');
+  $('.embedded-logfile').each(function (index, logFileElement) {
+    let content = logFileElement.dataset.content;
+    if (content === undefined) {
+      return;
+    }
+    if (string.length > 0) {
+      const lines = content.split(/\r?\n/);
+      const wanted = [];
+      for (let i = 0; i < lines.length; i++) {
+        if (regex) {
+          if (lines[i].match(regex)) {
+            wanted.push(lines[i]);
+          }
+          continue;
+        }
+        if (lines[i].includes(string)) {
+          wanted.push(lines[i]);
+        }
+      }
+      content = wanted.join('\n');
+      displaySearchInfo('Showing ' + wanted.length + ' / ' + lines.length + ' lines');
+    } else {
+      displaySearchInfo('');
+    }
+    logFileElement.innerHTML = ansiToHtml(content);
+  });
+  const fullCurrentUrl = window.location.href;
+  const currentUrl = fullCurrentUrl.split('#')[0];
+  const fragment = fullCurrentUrl.split('#')[1];
+  if (string.length > 0) {
+    window.location.href = `${currentUrl}#${string}`;
```

Comment:
```suggestion
    window.location.href = `${currentUrl}#filter=${string}`;
```

Reply:
No, this won't work well when we add a new parameter here. The string needs to be encoded.